File: code_examples/3-Structured_Problem_Solving/17-Continue/CSharp/main.cs
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a for loop
*/

using System;

/*
 * The Program class
*/
class Program {
    static void Main() {
        // this function uses a do ... while loop

        int positiveInteger;

        // input
        Console.Write("Enter how many times to repeat: ");
        positiveInteger = Convert.ToInt32(Console.ReadLine());
        Console.WriteLine();

        // process & output
        while (positiveInteger >= 0) {
            // yes, this is the exception on placing the counter at the top
            // if you did not, then there would be an infinit loop

            positiveInteger--;
            if (positiveInteger + 1 == 5) {
                continue;
            }
            Console.WriteLine("Current variable value: " + (positiveInteger + 1));
        }

        Console.WriteLine("\nDone.");
    }
}

File: code_examples/3-Structured_Problem_Solving/17-Continue/C/main.c
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program uses a for loop

#include <stdio.h>

int main() {
    // this function uses a for loop
    int positiveInteger;

    // input
    printf("Enter how many times to repeat: ");
    scanf("%d", &positiveInteger);
    printf("\n");

    // process & output
    while (positiveInteger >= 0) {
        // yes, this is the exception on placing the counter at the top
        // if you did not, then there would be an infinit loop

        positiveInteger--;
        if (positiveInteger == 5) {
            continue;
        }
        printf("Current variable value: %d\n", positiveInteger + 1);
    }

    printf("\nDone.\n");
    return 0;
}
